### PR TITLE
Promote API version 2 to supported

### PR DESCRIPTION
**Pull request by Bronek**
## High Level Overview of Change

This PR version essentially bumps `apiMaximumSupportedVersion` to 2, whilst ensuring that the command line interface (both `rippled` and the one used in unit tests) stays at version 1. We plan to bump the command line interface to API version 2, but not in this PR (probably also not in the nearest release). We also bump `apiBetaVersion` to 3 and add `apiCommandLineVersion`, used by both command line parsing and these unit tests which rely on command line parsing.

Note, this change removes remnants of a non-functional method `tx_account`. The method has no implementation and no mapping in `Handler.cpp`, however until this PR it did have command line parsing (only used in unit tests and otherwise useless). This is now removed.

Large diff is result of the switch to `forAllApiVersions` in `RPCall_test.cpp`. This test was failing because of hardcoded dependency on `apiMaximumSupportedVersion` (which before this change was incidentally the same as `apiMinimumSupportedVersion`), expressed both as `%MAX_API_VER%` and as a conditional on expected output (near lines 1353 - 1369). Since change was needed to fix the failing test, and large diff would be unavoidable to remove the misleading `%MAX_API_VER%` name, I decided to do it whilst improving test design and coverage.

### Type of Change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Refactor (non-breaking change that only restructures code)
- [ ] Tests (you added tests for code that already exists, or your new feature included in this PR)
- [ ] Documentation update
- [ ] Chore (no impact to binary, e.g. `.gitignore`, formatting, dropping support for older tooling)
- [ ] Release

### CODE REVIEW

**Comment by Bronek** [comment]
The second commit https://github.com/XRPLF/rippled/pull/4803/commits/35b1da735cd6fd405fd8ae78477e2939ecc4d0b4 contains a pretty bad attempt to pass `apiVersion` into `RPCParser` ; I will rewrite it.

**Review by ximinez — CHANGES_REQUESTED** [review]
Looks good overall, but I do have some feedback.

File: src/ripple/net/impl/RPCCall.cpp [hunk 1]
```diff
@@ -1673,7 +1603,7 @@ fromCommandLine(
     Logs& logs)
 {
     auto const result =
-        rpcClient(vCmd, config, logs, RPC::apiMaximumSupportedVersion);
+        rpcClient(vCmd, config, logs, RPC::apiMinimumSupportedVersion);
```
[the remark this answers is not in the document]
Comment (Bronek) [reply src/ripple/net/impl/RPCCall.cpp:1606]: oh I like this idea. For now this will be explicitly 1 because we do not want to push forward too fast (in this release), but to be updated in a separate PR soon after

File: src/test/jtx/Env.h [hunk 2]
```diff
@@ -703,7 +740,9 @@ Env::rpc(
     Args&&... args)
 {
     return do_rpc(
-        std::vector<std::string>{cmd, std::forward<Args>(args)...}, headers);
+        std::vector<std::string>{cmd, std::forward<Args>(args)...},
+        RPC::apiMinimumSupportedVersion,
+        headers);
```
[the remark this answers is not in the document]
Comment (Bronek) [reply src/test/jtx/Env.h:745]: Nope, that's a design mistake that took me whole day to unwind, and only partially. We want all unit tests for command line to pass `forAllApiVersions` and, when this happens, remove this default entirely.

In other words, we want to be able to switch `apiCommandLineVersion` which you proposed at will, knowing that that the newly selected (and any other) version is already covered by unit tests which do pass, before we even make such change.

File: src/test/jtx/Env.h [hunk 3]
```diff
@@ -279,6 +279,17 @@ class Env
         The command is examined and used to build
         the correct JSON as per the arguments.
     */
+    template <class... Args>
+    Json::Value
+    rpc(std::unordered_map<std::string, std::string> const& headers,
+        unsigned apiVersion,
+        std::string const& cmd,
+        Args&&... args);
```
[the remark this answers is not in the document]
Comment (Bronek) [reply src/test/jtx/Env.h:287]: I think you are right, thanks

File: src/test/jtx/Env.h (hunk 2 above)
[the remark this answers is not in the document]
Comment (Bronek) [reply src/test/jtx/Env.h]: Huh, there are too many tests to use `forAllApiVersions` everywhere. So majority of tests use exactly the same version as the one used by `RPCCall` i.e. `apiCommandLineVersion` - this will give us the minimum of test coverage I was after. I will update this line accordingly.

**Review by ximinez — APPROVED** [review]

**Review by seelabs — APPROVED** [review]
:+1: LGTM

File: Builds/CMake/RippledCompiler.cmake [hunk 4]
```diff
@@ -131,7 +131,16 @@ else ()
       >)
 endif ()
 
-if (use_gold AND is_gcc)
+if (use_mold)
```
Comment (ckniffen) [inline Builds/CMake/RippledCompiler.cmake:134]: Were these changes required to allow version 2?
